Reduce the number of cppcheck-related issues

vitalii-komisarenko opened the pull request:
The list of existing cppcheck issues is almost 700 entries long.
It is hard to find the important ones because there are too many of them.

The purpose of this pull request is to reduce their number to make this list more manageable.
And it is nice to have code that follows code style.

vitalii-komisarenko committed:
cppcheck: controller: db: pass arguments by const reference

Codestyle requires developers to pass function arguments bigger than 4 bytes by reference.
This commit fixes these code style issues for db.cpp and db.h

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: controller db: remove const&-related

Files db.h and db.cpp of the controller has been changed to pass values by const reference
This commit removes them from list of existing issues since they have been fixed

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck: pass arguments by reference for bml_internal (controller)

Pass arguments by reference where needed for bml_internal.{cpp,h}

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: bml_internal: remove const& issues

Since "pass by value" issues were fixed for bml_internal{.cpp,.h}
there is no need to whitelist them in cppcheck_existing_issues.txt

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck: beerocks_string_utils: pass checks

Fix cppcheck issues in beerocks_string_utils.h and beerocks_string_utils.cpp

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: remove beerocks_string_utils issues

Remove beerocks_string_utils.{cpp,h} from the list of existing issues
because all their issues have been fixed

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck: beerocks_slave_main: fix cppcheck issues

Fixed cppcheck issues in beerocks_slave_main.cpp

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: remove beerocks_slave_main

Remove beerocks_slave_main-related issues from the list of known issues
because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: monitor_db: fix cppcheck issues

Fix cppcheck issues in monitor_db.cpp and monitor_db.h

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: remove monitor_db (agent)

Remove monitor_db-related cppcheck issues from list of know issues
because they have been fixed

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks_utils: make code pass cppcheck

Make beerocks_utils.cpp and beerocks_utils.h pass cppcheck check.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: remove beerocks_utils

Remove beerocks_utils-related issues from the list of know issues
because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
beerocks_os_utils: make code cppcheck-compliant

Make beerocks_os_utils.cpp and beerocks_os_utils.h pass cppcheck check.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
cppcheck_existing_issues: remove beerocks_os_utils

Remove beerocks_os_utils-related cppcheck issues from the list of known issues
because they have been fixed.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

arnout commented on common/beerocks/bcl/source/beerocks_string_utils.cpp:
```diff
@@ -114,7 +114,6 @@ std::string string_utils::int_to_hex_string(const unsigned int integer,
     // for example: int_to_hex_string(255, 4) -> "00ff"
     //              int_to_hex_string(255, 1) -> "ff"
 
-    std::string return_string;
```
I really liked your first few commits: they are really clear and though they are large, it's trivial to check that they do the right thing.

Here, however, it's a lot more difficult, because there are two different kind of changes in the same commit, and the commit message doesn't describe them. So I'd like this to be split into two commits: the first one that does the string replacements, and the second one that fixes this issue (which I assume is an unused variable).

arnout commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -139,7 +139,7 @@ static std::string get_sta_iface_from_hostap_iface(const std::string &hostap_ifa
 }
 
-static void fill_son_slave_config(beerocks::config_file::sConfigSlave &beerocks_slave_conf,
+static void fill_son_slave_config(const beerocks::config_file::sConfigSlave &beerocks_slave_conf,
```
This is also a const-ref fix so please reuse the same commit message as in the first two.

arnout commented on agent/src/beerocks/slave/beerocks_slave_main.cpp:
```diff
@@ -206,11 +206,10 @@ static void son_slave_watchdog(const std::string &beerocks_slave_temp_path,
+                            int argc, char *argv[])
 {
     std::string name = std::string("system_hang_test");
-    std::ofstream versionfile;
```
This one again is separate.

You could combine a few unused variable fixes in a single commit, if you want to reduce the number of commits.

arnout commented on common/beerocks/bcl/source/beerocks_utils.cpp:
```diff
@@ -215,7 +215,7 @@ std::string utils::convert_channel_ext_above_to_string(bool channel_ext_above_se
 {
     std::vector<uint8_t>::iterator it;
-    for (it = list.begin(); it != list.end(); it++) {
+    for (it = list.begin(); it != list.end(); ++it) {
```
For this one, a better fix is to remove the function entirely since it's not used.

Or failing that, use ranged loop:
```
    for (element : list) {
```

RanRegev replied:
For my understanding this function might not do what you want:
- it does not copies zeros from the source (list), but it does not verify there are no zeros in the destination (ret). So you might end with zeros at ret.
if order does not matter, the schema of the function should be:
```
ret.insert(list.begin(), list.end());
std::sort(ret);
std::remove(ret.begin(),ret.end(),0);
std::unique(ret.begin(),ret.end());
```
Using merely algorithms from std. No raw loops.
example reference: https://youtu.be/W2tWOdzgXHA

vitalii-komisarenko replied:
@RanRegev , I prefer @arnout 's approach, not only because it is easier, but also because this function is too bad to keep:
* it is unused
* use case too niche to happen often
* code style does not correspond to current guidelines
* the name is misleading (should be something like `merge_non_zero_values_from_list`)

arnout reviewed (approved):
Excellent PR! I like it when things are improved!

I made a few comments about commit splitting, but it's not worth spending the time to actually do that - it's more a suggestion for the future.

The only thing I would like to change is the merge_list function.

RanRegev commented on agent/src/beerocks/monitor/monitor_db.h:
```diff
@@ -23,7 +23,10 @@ namespace son {
 public:
-    monitor_sta_node(const int8_t vap_id_, const std::string mac_) { vap_id = vap_id_, mac = mac_; }
+    monitor_sta_node(const int8_t vap_id_, const std::string &mac_)
+        : vap_id(vap_id_), mac(mac_), m_sta_stats()
```
No need to call the default constructor explicitly. I am referring to the explicit call to `m_sta_stats()`
references -
standard: http://eel.is/c++draft/class.base.init#9
godbolt: https://godbolt.org/z/4YEwWS
Also apply for more default initialized members along the code.

vitalii-komisarenko replied:
cppcheck actually complains about uninitialized members in the constructor:
`'/builds/prpl-foundation/prplMesh/agent/src/beerocks/monitor/monitor_db.h: warning: Member variable 'monitor_sta_node::m_sta_stats' is not initialized in the constructor. [uninitMemberVar]    monitor_sta_node(const int8_t vap_id_, const std::string mac_) { vap_id = vap_id_, mac = mac_; }`

rmelotte replied:
Actually, I think cppcheck is right here.
`m_sta_stats` is of type SStaStats, which comes from `common/beerocks/bwl/include/bwl/mon_wlan_hal_types.h` (and there is an other, different definition in `agent/src/beerocks/monitor/monitor_db.h`).

The struct is defined as follow:
```
struct SStaStats {
    float rx_rssi_watt;
    uint8_t rx_rssi_watt_samples_cnt;
    float rx_snr_watt;
    uint8_t rx_snr_watt_samples_cnt;
    // int8_t   rx_rssi_prev=beerocks::RSSI_INVALID;
    // int8_t   rx_rssi_curr=beerocks::RSSI_INVALID;
    uint16_t tx_phy_rate_100kb;
    // uint16_t tx_phy_rate_100kb_avg;
    // uint16_t tx_phy_rate_100kb_min;
    // uint16_t tx_phy_rate_100kb_acc;
    uint16_t rx_phy_rate_100kb;
    // uint16_t rx_phy_rate_100kb_avg;
    // uint16_t rx_phy_rate_100kb_min;
    // uint16_t rx_phy_rate_100kb_acc;
    uint64_t tx_bytes_cnt;
    uint64_t rx_bytes_cnt;
    uint64_t tx_packets_cnt;
    uint64_t rx_packets_cnt;
    uint32_t tx_packets;
    uint32_t tx_bytes;
    uint32_t rx_packets;
    uint32_t rx_bytes;
    uint32_t retrans_count;
    // uint8_t  tx_load_percent_curr=0;
    // uint8_t  tx_load_percent_prev=0;
    // uint8_t  rx_load_percent_curr=0;
    // uint8_t  rx_load_percent_prev=0;
};
```

If you don't initialize m_sta_stats explicitly, all of these will be default-initialized, so they will have a random value.

I adapted your example on godbolt: https://godbolt.org/z/LWmccS

RanRegev replied:
yes, I you are correct.
No reason to default initialize this struct unless specifically requested.
My mistake.
standard: http://eel.is/c++draft/dcl.init#7.3

vitalii-komisarenko committed:
beerocks_utils: remove `merge_list` function

Function `merge_list` is a function that appends non-zero numbers to a vector.
It is removed in this commit because:
* it is not used at the moment
* its use case is too niche, thus improbable in the future
* its code does not correspond to our coding style guidelines
* its name is misleading, since it "merges" only non-zero value

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

rmelotte reviewed (approved):
It solves so many cppcheck issues that Github does not load the diff by default.
Nice! :-)
